pkg/testutils/kubernetes/kubectl.go

```diff
@@ -0,0 +1,173 @@
+package kubernetes
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+
+	"github.com/operator-framework/ansible-operator-plugins/pkg/testutils/command"
+)
+
+// Kubectl represents a command line utility that is used for interacting with a Kubernetes cluster
+type Kubectl interface {
```

Comment:
>If we construct a command with Kustomize and ansible plugin in the project itself and build a binary, can't we still use kubebuilder's test context? 

I _think_ we could. I decided to go a different route here that is, IMO, easier to understand than using the Kubebuilder `TestContext` and is better tailored to the phase 1.5 plugin implementation. That being said, if we _really_ wanted to use the `TestContext` I can attempt to convert to that instead.

The whole goal here is that the Kubebuilder `TestContext` has a lot of utility/helper functionality tied to it that seems unnecessary to require the creation of the test context (instead they can simply be exported functions with the necessary parameters as is done here). This makes it easier, IMO, to follow and use these helper functions. It also seems a bit unnecessary to require the building and shipping of a binary for the phase 1.5 plugin just to enable testing purposes (especially when the goal of the phase 1.5 plugin is that it is a library consumed by Operator-SDK for now) and that led me to creating this sample generator logic that allows the creation and usage of a Kubebuilder `cli.Cli` object instead of being required to use a built binary.

Reply:
see https://github.com/operator-framework/ansible-operator-plugins/pull/4#discussion_r1300612715